apps/accounts/forms/update.py

```diff
@@ -129,3 +133,30 @@ def save(self, user, *args, **kwargs):
             }
             rollbar.report_exc_info(request=request, extra_data=extra_data)
         return user
+
+
+class TimeZoneForm(AbstractModelInstanceUpdateForm):
+    class Meta:
+        model = UserProfile
+        fields = (
+            'timezone',
+        )
+
+    def __init__(self, instance, *args, **kwargs):
+        """Initialization for TimeZoneForm.
+        """
+        user = instance
+        user_profile = user.profile
+        super(TimeZoneForm, self).__init__(user, *args, **kwargs)
+
+        self.fields['timezone'].initial = user_profile.timezone
+        self.fields['timezone'].label = ''
+
+    def save(self, request, *args, **kwargs):
+        user = super(TimeZoneForm, self).save(request, *args, **kwargs)
+        user_profile = user.profile
+        timezone = request.POST.get('timezone', '')
+        default_timezone = htk_setting('HTK_DEFAULT_TIMEZONE', 'UTC')
+        user_profile.timezone = timezone or default_timezone
```

Comment:
```
default_timezone = ...
timezone = request.POST.get('timezone', default_timezone)
user_profile.timezone = timezone
```

Reply:
The empty(-----) also one of the options in the timezone field of the UserProfile, 
![image](https://user-images.githubusercontent.com/55696493/110653482-4cbdd280-81e3-11eb-9080-8dadd5eccb0a.png)

which is due to blank attribute is set to true in the following line, 
https://github.com/hacktoolkit/django-htk/blob/4441c12d1b5e8592a4039973754d33969e996cc3/apps/accounts/models.py#L69

If we keep the above code (https://github.com/hacktoolkit/django-htk/blob/44ea9f6c244bcfc5b50aee858c9217e43085d265/apps/accounts/forms/update.py#L158-L160) then it will set it to default timezone when the timezone is empty otherwise, We need to change to the field to required by setting blank to False here  https://github.com/hacktoolkit/django-htk/blob/4441c12d1b5e8592a4039973754d33969e996cc3/apps/accounts/models.py#L69


Shall I keep this code?